x/tokenomics/keeper/proof_requirement_test.go

```diff
@@ -0,0 +1,61 @@
+package keeper_test
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"cosmossdk.io/log"
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/testutil/keeper"
+	tetsproof "github.com/pokt-network/poktroll/testutil/proof"
+	"github.com/pokt-network/poktroll/testutil/sample"
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+)
+
+func init() {
+	rand.Seed(0)
+}
+
+func TestKeeper_IsProofRequired(t *testing.T) {
+	// Set expectedCompute units to be below the proof requirement threshold to only
+	// exercise the probabilistic branch of the #isProofRequired() logic.
+	expectedComputeUnits := prooftypes.DefaultProofRequirementThreshold - 1
+	keepers, ctx := keeper.NewTokenomicsModuleKeepers(t, log.NewNopLogger())
+	sdkCtx := cosmostypes.UnwrapSDKContext(ctx)
+
+	var (
+		// Because this test is deterministic & this sample size is known to be
+		// sufficient, it doest not need to be calculated.
+		sampleSize  = 1500
+		samples     = make(map[bool]int64)
+		probability = prooftypes.DefaultProofRequestProbability
+		tolerance   = 0.01
+	)
+
+	for i := 0; i < sampleSize; i++ {
+		claim := tetsproof.ClaimWithRandomHash(t, sample.AccAddress(), sample.AccAddress(), expectedComputeUnits)
+
+		isRequired, err := keepers.Keeper.IsProofRequiredForClaim(sdkCtx, &claim)
+		require.NoError(t, err)
+
+		samples[isRequired]++
+	}
+
+	// Check that the number of samples for each outcome is within the expected range.
+	for outcome, count := range samples {
```

Comment:
IMO, removing the loop and deal with the `true` and `false` cases explicitly would be more readable.

Reply:
Another great suggestion. :sweat_smile: